Rover/sailboat.cpp

```diff
@@ -552,3 +585,111 @@ bool Sailboat::motor_assist_low_wind() const
             rover.g2.windvane.wind_speed_enabled() &&
             (rover.g2.windvane.get_true_wind_speed() < sail_windspeed_min));
 }
+
+bool Sailboat::in_irons()
```

Comment:
This is quite a big function, and its called in several places, I think it would be better to check one time, maybe in `use_indirect_route` then we can just return the result of that check for other users.